### PR TITLE
Include Garment Support for Custom Garments

### PR DESCRIPTION
**Pull request by engres01**
Include Garment Support for Custom Garments

Implemented:
- Write to custom accessors in glb when exporting garment mesh
- Write to Garment Support parameters when importing mesh with garment support data

### CODE REVIEW

File: WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs [hunk 1]
```diff
@@ -1726,5 +1731,129 @@ private static void UpdateSkinningParamCloth(ref List<RawMeshContainer> meshes,
                 }
             }
         }
+
+        private static void UpdateGarmentSupportParameters(List<RawMeshContainer> meshes, ref CR2WFile cr2w)
```
Comment (rfuzzo) [inline WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs:1735]: Does this need to be ref?

File: WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs [hunk 2]
```diff
@@ -1726,5 +1731,129 @@ private static void UpdateSkinningParamCloth(ref List<RawMeshContainer> meshes,
...
+        {
+            var cMesh = (CMesh)cr2w.RootChunk;
```
Comment (rfuzzo) [inline WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs:1737]: This shouldn't be a cstyle cast. Better to use pattern matching: if (cr2w.Root is CMesh cMesh)

File: WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs [hunk 3]
```diff
@@ -1726,5 +1731,129 @@ private static void UpdateSkinningParamCloth(ref List<RawMeshContainer> meshes,
...
+
+        private static meshMeshParamGarmentSupport GetParameter_meshMeshParamGarmentSupport(CMesh cMesh)
+        {
+            var garmentMeshBlob = cMesh.Parameters.FirstOrDefault(x => x.Chunk is meshMeshParamGarmentSupport);
+            if (garmentMeshBlob == null)
+            {
+                garmentMeshBlob = new CHandle<meshMeshParameter> { Chunk = new meshMeshParamGarmentSupport() };
+                cMesh.Parameters.Add(garmentMeshBlob);
+            }
+            var garmentMeshBlobChunk = (meshMeshParamGarmentSupport)garmentMeshBlob.Chunk;
```
Comment (rfuzzo) [inline WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs:1764]: Also use pattern matching here

File: WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs [hunk 4]
```diff
@@ -1726,5 +1731,129 @@ private static void UpdateSkinningParamCloth(ref List<RawMeshContainer> meshes,
...
+
+        private static garmentMeshParamGarment GetParameter_garmentMeshParamGarment(CMesh cMesh)
+        {
+            var garmentBlob = cMesh.Parameters.FirstOrDefault(x => x.Chunk is garmentMeshParamGarment);
+            if (garmentBlob == null)
+            {
+                garmentBlob = new CHandle<meshMeshParameter> { Chunk = new garmentMeshParamGarment() };
+                cMesh.Parameters.Add(garmentBlob);
+            }
+            var garmentBlobChunk = (garmentMeshParamGarment)garmentBlob.Chunk;
```
Comment (rfuzzo) [inline WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs:1779]: No cstyle casting here please

**Review by rfuzzo — CHANGES_REQUESTED** [review]
Looks good to me, just Somme code style things so that the compiler is happy

File: WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs (hunk 1 above)
Comment (engres01) [reply WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs]: Removed ref from parameter

File: WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs (hunk 2 above)
Comment (engres01) [reply WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs]: Changed to pattern matching

File: WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs (hunk 3 above)
Comment (engres01) [reply WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs]: Changed to pattern matching

File: WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs (hunk 4 above)
Comment (engres01) [reply WolvenKit.Modkit/RED4/Tools/MeshImportTools.cs]: Changed to pattern matching

**Review by rfuzzo — APPROVED** [review]
thanks, this looks good to me

**Pull request merged**